pkg/convert/pot_test.go

```diff
@@ -0,0 +1,317 @@
+		{
+			name: "When fuzzy values are missing",
+			input: model.Messages{
+				Language: language.English,
+				Messages: []model.Message{
+					{ID: "Hello, world!", Message: "Bonjour le monde!", Description: "A simple greeting"},
+					{ID: "Goodbye!", Message: "Au revoir!", Description: "A farewell"},
+				},
+			},
+			expected: []byte(`"Language: en
+#. A simple greeting
+msgid "Hello, world!"
+msgstr "Bonjour le monde!"
+#. A farewell
+msgid "Goodbye!"
+msgstr "Au revoir!"
+`),
+		},
+		{
+			name: "When description value is missing",
+			input: model.Messages{
+				Language: language.English,
+				Messages: []model.Message{
+					{ID: "Hello, world!", Message: "Bonjour le monde!", Fuzzy: true},
+					{ID: "Goodbye!", Message: "Au revoir!", Description: "A farewell", Fuzzy: true},
+				},
+			},
+			expected: []byte(`"Language: en
+#, fuzzy
+msgid "Hello, world!"
+msgstr "Bonjour le monde!"
+#. A farewell
+#, fuzzy
+msgid "Goodbye!"
+msgstr "Au revoir!"
+`),
+		},
+		{
+			name: "When description and fuzzy values are missing",
+			input: model.Messages{
+				Language: language.English,
+				Messages: []model.Message{
+					{ID: "Hello, world!", Message: "Bonjour le monde!"},
+					{ID: "Goodbye!", Message: "Au revoir!"},
+				},
+			},
+			expected: []byte(`"Language: en
+msgid "Hello, world!"
+msgstr "Bonjour le monde!"
+msgid "Goodbye!"
+msgstr "Au revoir!"
+`),
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			result, err := ToPot(tt.input)
+
+			if !assert.NoError(t, err) {
+				return
+			}
+			fmt.Printf("exp: %v\n", string(tt.expected))
+			fmt.Printf("res: %v\n", string(result))
+			assert.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+func TestFromPot(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name        string
+		expected    model.Messages
+		expectedErr error
+		input       []byte
+	}{
+		{
+			name: "Valid input",
+			input: []byte(`# Language: en-US
+							#. "a greeting"
+							#, ""
+							msgid "Hello"
+							msgstr "Hello, world!"
+							
+							#. "a farewell"
+							#, "fuzzy"
+							msgid "Goodbye"
+							msgstr "Goodbye, world!"
+			`),
+			expected: model.Messages{
+				Language: language.Make("en-US"),
+				Messages: []model.Message{
+					{
+						ID:          "Hello",
+						Message:     "Hello, world!",
+						Description: "a greeting",
+						Fuzzy:       false,
+					},
+					{
+						ID:          "Goodbye",
+						Message:     "Goodbye, world!",
+						Description: "a farewell",
+						Fuzzy:       true,
+					},
+				},
+			},
+		},
+		{
+			name: "Invalid input",
+			input: []byte(`# Language: en-US
+							#. "a greeting"
+							#, ""
+							msgid 323344
+			`),
+			expectedErr: fmt.Errorf("convert tokens to pot.Po: invalid po file: no messages found"),
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			result, err := FromPot(tt.input)
+			if err != nil {
```

Comment:
Do not use the actual value to drive assertions. ALWAYS use the expected values. This is a mistake made by so many developers. :)

In your case, if err is always nil, your assertion of `expected.Err` never takes place.

Reply:
![image](https://user-images.githubusercontent.com/69954124/221528353-3aab5c29-3fdd-4c7c-ac2d-d5f1be5d8c61.png)